### PR TITLE
K8SPSMDB-497: Take mongos into consideration for paused status

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPSMDB-497](https://jira.percona.com/browse/K8SPSMDB-497) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| operator-self-healing-chaos | failed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/713/commits/42e43246a29048e2d89cced9863fd1be17e3d4eb
image: `perconalab/percona-server-mongodb-operator:PR-713-42e43246`

File: pkg/controller/perconaservermongodb/status.go [hunk 1]
```diff
@@ -178,9 +178,9 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 	cr.Status.Host = host
 
 	switch {
-	case replsetsStopping > 0 || cr.Status.Mongos.Status == api.AppStateStopping || cr.ObjectMeta.DeletionTimestamp != nil:
+	case replsetsStopping > 0 || (cr.Spec.Sharding.Enabled && cr.Status.Mongos.Status == api.AppStateStopping) || cr.ObjectMeta.DeletionTimestamp != nil:
 		cr.Status.State = api.AppStateStopping
-	case replsetsPaused == len(repls) && cr.Status.Mongos.Status == api.AppStatePaused:
+	case replsetsPaused == len(repls) && (cr.Spec.Sharding.Enabled && cr.Status.Mongos.Status == api.AppStatePaused):
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/status.go:183]: if `cr.Spec.Sharding.Enabled == false` we will never get Paused state

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/status.go (hunk 1 above)
Comment (egegunes) [reply pkg/controller/perconaservermongodb/status.go]: https://github.com/percona/percona-server-mongodb-operator/pull/713/commits/205f268746056d69e107f12d322f6a3a7e648d16

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| one-pod | failed |
| storage | failed |
| upgrade-consistency | failed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| limits | failed |
| self-healing | failed |
| arbiter | failed |
| scheduled-backup | failed |
| scaling | failed |
| service-per-pod | failed |
| liveness | failed |
| self-healing-chaos | failed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | failed |
| security-context | failed |
| smart-update | failed |
| operator-self-healing | failed |
| pitr-sharded | passed |
| version-service | failed |
| rs-shard-migration | failed |
| operator-self-healing-chaos | failed |
| users | failed |
| data-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/713/commits/6aa1b8f025c1f688a7e8ef3469151debb7d702b4
image: `perconalab/percona-server-mongodb-operator:PR-713-6aa1b8f0`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| service-per-pod | failed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade | passed |
| liveness | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/713/commits/205f268746056d69e107f12d322f6a3a7e648d16
image: `perconalab/percona-server-mongodb-operator:PR-713-205f2687`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| one-pod | passed |
| init-deploy | passed |
| monitoring-2-0 | passed |
| limits | passed |
| arbiter | passed |
| scaling | passed |
| service-per-pod | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/713/commits/205f268746056d69e107f12d322f6a3a7e648d16
image: `perconalab/percona-server-mongodb-operator:PR-713-205f2687`

**Review by qJkee — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**